src/profile-logic/marker-data.js

```diff
@@ -392,16 +392,30 @@ export function filterRawMarkerTableToRange(
 ): RawMarkerTable {
   const newMarkerTable = getEmptyRawMarkerTable();
 
+  const filteredMarkerIndexesIter = filterRawMarkerTableIndexesToRange(
+    markers,
+    rangeStart,
+    rangeEnd
+  );
+
+  for (const index of filteredMarkerIndexesIter) {
+    newMarkerTable.time.push(markers.time[index]);
+    newMarkerTable.name.push(markers.name[index]);
+    newMarkerTable.data.push(markers.data[index]);
+    newMarkerTable.length++;
+  }
+  return newMarkerTable;
+}
+
+export function* filterRawMarkerTableIndexesToRange(
```

Comment:
At least the name should better state that it's a generator. Maybe rename to `filterRawMarkerTableToRangeIndexGenerator`

Reply:
I know it's not commonly used language feature, but I think it's good to use that feature here to avoid generating an intermediate marker table. I'm adding some comments to explain it. Also Changed the name to `filterRawMarkerTableToRangeIndexGenerator`

Reply:
Thank you for the explanation and additional comments for this.